### PR TITLE
Add support for Python 3.7

### PR DESCRIPTION
**Pull request by hugovk**
### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Add support for Python 3.7
- A small workaround is needed for Travis CI, see https://github.com/travis-ci/travis-ci/issues/9815

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=615) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=615) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=615) it.</sub>

File: setup.py [hunk 1]
```diff
@@ -32,13 +32,17 @@ def getRequires():
     description='SendGrid library for Python',
     long_description=long_description,
     install_requires=getRequires(),
+    python_requires='>=2.6, !=3.0.*, !=3.1.*',
     classifiers=[
+        'Programming Language :: Python :: 2',
```
[the remark this answers is not in the document]
Comment (hugovk) [reply setup.py]: This is usually used to indicate at least *some* version of Python 2 is used, not necessarily all 2.0+.

But there's no hard and fast rule. I'll be happy to remove it when I'm next at the keyboard, if you prefer.

[the remark this answers is not in the document]
Comment (hugovk) [reply setup.py]: Removed.

**Comment by thinkingserious** [comment]
Hello @hugovk,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Can you please move this to the v4 branch? Thanks!

**Comment by hugovk** [comment]
Changing the base branch didn't work very smoothly! Closing this in favour of https://github.com/sendgrid/sendgrid-python/pull/637.

**Pull request closed without merging**